internal/lookoutv2/model/model.go

```diff
@@ -42,16 +42,52 @@ type Job struct {
 	CancelReason       *string
 }
 
+// PostgreSQLTime is a wrapper around time.Time that converts to UTC when
+// deserializing from JSON.
+//
+// It exists to work around the following issue:
+//
+//  1. PostgreSQL serializes UTC timestamps within a JSON object in the format
+//     "2023-11-03T09:10:42.201577+00:00"; in particular, this format uses a
+//     timezone offset instead of "Z" to indicate UTC.
+//  2. When deserializing this UTC timestamp, Go sets the location of the
+//     resulting time.Time value to "local".
+//  3. Tests compare timestamps with == instead of Equal, which means that two
+//     time.Time values with different locations are not considered equal
+//     (even if they represent the same instants in time).
+type PostgreSQLTime struct {
+	Time time.Time
+}
+
+func NewPostgreSQLTime(t *time.Time) *PostgreSQLTime {
+	if t == nil {
+		return nil
+	}
+	return &PostgreSQLTime{Time: *t}
+}
+
+func (t PostgreSQLTime) MarshalJSON() ([]byte, error) {
```

Comment:
I  presume this is  is needed when you have timestamps int he jsonb column?  If so is it ok that this isn't nil safe? (c.f. `NewPostgreSQLTime` which will happily return you  a nil).

Reply:
In a previous version of this code `MarshalJSON` operated on `*PostgreSQLTime` (and blindly dereferenced `t`); in the version you're commenting on it operates on `PostgreSQLTime`, because I had similar concerns. However, both versions are fine (`json.Marshal` doesn't call `MarshalJSON` on nil pointers): https://go.dev/play/p/JpvUKtlWILQ

Reply:
We discussed the analogous question for `UnmarshalJSON` offline; the upshot is that `json.Unmarshal` also guarantees that the pointer it calls `UnmarshalJSON` on is not nil (I modeled this after the [implementation of `UnmarshalJSON` for `time.Time`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/time/time.go;l=1375-1389), which also just dereferences the pointer).